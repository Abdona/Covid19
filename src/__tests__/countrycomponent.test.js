import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Mockedcountry from '../mocks/mockedCountry';
import '@testing-library/jest-dom';

describe('Test Home page Components', () => {
  it('test HomePage function component renders without error', () => {
    const divElement = document.createElement('div');
    ReactDOM.render(<Mockedcountry />, divElement);
  });

  it('render correctly', () => {
    const { getByTestId } = render(<Mockedcountry country="Egypt" today_confirmed="239" />);
    expect(getByTestId('country')).toHaveTextContent('Egypt239');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Mockedcountry />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
