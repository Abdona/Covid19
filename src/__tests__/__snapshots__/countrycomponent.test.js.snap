// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page Components matches snapshot 1`] = `
<div
  className="country2"
  data-testid="country"
>
  <div
    className="imgcont"
  >
    <img
      src="country.svg"
    />
    
  </div>
  <div
    className="textcont"
  >
    <h1 />
    <h3 />
  </div>
</div>
`;
